compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1086,4 +1086,37 @@ object messages {
           |""".stripMargin
   }
 
+  case class MixedLeftAndRightAssociativeOps(op1: Name, op2: Name, op2LeftAssoc: Boolean)(implicit ctx: Context)
+    extends Message(41) {
+    val kind = "Syntax"
+    val op1Asso = if (op2LeftAssoc) "which is right-associative" else "which is left-associative"
+    val op2Asso = if (op2LeftAssoc) "which is left-associative" else "which is right-associative"
+    val msg = s"${hl"`${op1}`"} (${op1Asso}) and ${hl"`${op2}`"} ($op2Asso) have same precedence and may not be mixed"
+    val explanation =
+      s"""|The operators ${hl"${op1}"} and ${hl"${op2}"} are used as infix operators in the same expression,
```

Comment:
why not use the `hl` interpolator? You still have to interpolate values for them to be highlighted. If there is some interpolated part of the message that gets highlighted incorrectly, you can wrap it in `NoColor(..)` or some other "color" that you find more suitable.

Anyway, choose the lesser evil - if `hl` requires more boilerplate, then of course `s` is preferable.